### PR TITLE
Queue up event reporting task if Library isn't configured

### PR DESCRIPTION
**Pull request by najmsheikh**
### Goals :soccer:
To ensure that deeplink events aren't dropped if the Library isn't configured at the time `trackIncomingIntent(~)` is called.

### Implementation :construction:
I keep track of the pending `Task`s that were to be invoked if the submission occurred prior to the `configure(~)` call. Once the application ID is provided, I immediately submit the pending tasks.
N.B. to retain parity with iOS, this PR only queues up the deeplink event reporting task.

### CODE REVIEW

File: button-merchant/src/test/java/com/usebutton/merchant/ButtonRepositoryImplTest.java [hunk 1]
```diff
@@ -73,6 +75,38 @@ public void setApplicationId_provideToButtonApi() {
...
+        buttonRepository.reportEvent(mock(DeviceManager.class), mock(Features.class),
+                mock(Event.class));
+        buttonRepository.reportEvent(mock(DeviceManager.class), mock(Features.class),
+                mock(Event.class));
+
+        buttonRepository.setApplicationId("valid_application_id");
+        buttonRepository.setApplicationId("valid_application_id");
+        buttonRepository.setApplicationId("valid_application_id");
+        buttonRepository.setApplicationId("valid_application_id");
+        buttonRepository.setApplicationId("valid_application_id");
```
Comment (paultopher) [inline button-merchant/src/test/java/com/usebutton/merchant/ButtonRepositoryImplTest.java:105]: That's a lot of valid app ids

**Review by paultopher — APPROVED** [review]

File: button-merchant/src/main/java/com/usebutton/merchant/ButtonRepositoryImpl.java [hunk 2]
```diff
@@ -72,6 +75,11 @@ static ButtonRepository getInstance(ButtonApi buttonApi, PersistenceManager pers
     public void setApplicationId(String applicationId) {
         this.applicationId = applicationId;
         buttonApi.setApplicationId(applicationId);
+
+        for (Task<?> task : pendingTasks) {
+            executorService.submit(task);
```
Comment (wessmith) [inline button-merchant/src/main/java/com/usebutton/merchant/ButtonRepositoryImpl.java:80]: These tasks will fail without an appId. I don't see where you're adding the appId to the requests once we've received it. I just realized I have the [same issue over here](https://github.com/button/button-merchant-ios/pull/61).

Comment (wessmith) [reply button-merchant/src/main/java/com/usebutton/merchant/ButtonRepositoryImpl.java:80]: [Added on iOS](https://github.com/button/button-merchant-ios/pull/61/commits/3fc3df19f6b8b93f2f431604eec7cf2d938d5328)

Comment (najmsheikh) [reply button-merchant/src/main/java/com/usebutton/merchant/ButtonRepositoryImpl.java:80]: @wessmith The bug you mentioned isn't likely to occur here. The application_id is appended in `ConnectionManager`. That class only comes into play once the `Task` has been submitted -- which only occurs after `configure(~)`.

**Pull request merged**